src/OrbitGl/MainThreadExecutor.h

```diff
@@ -65,6 +67,142 @@ class MainThreadExecutor {
     return future;
   }
 
+  // ScheduleAfter schedules the continuation `functor` to be executed
+  // on `*this` after `future` has completed.
+  //
+  // Note: The continuation is only executed if `*this` is still alive when `future` completes.
+  template <typename F>
+  [[nodiscard]] auto ScheduleAfter(const orbit_base::Future<void>& future, F&& functor)
+      -> orbit_base::Future<std::decay_t<decltype(functor())>> {
+    CHECK(future.IsValid());
+
+    using ReturnType = std::decay_t<decltype(functor())>;
+
+    // Since std::function needs to be copyable, promise also needs to be copyable.
+    auto promise = std::make_shared<orbit_base::Promise<ReturnType>>();
+    orbit_base::Future<ReturnType> resulting_future = promise->GetFuture();
+
+    // Since MSVC2017 crashes with a compiler error when using `if constexpr` inside of lambda, we
```

Comment:
I'm not into the level of details about the crashes in MSVC2017, but what do you think about creating an internal bug to try to merge some of them together in the future (or a TODO)? Maybe you already thought a lot here, in this case please disregard my comment.

Reply:
I already have a fix lined up which reduces `ScheduleAfter` to a single function with no `if constexpr`.